src/Flysystem/FlysystemAssetStore.php

```diff
@@ -597,10 +598,15 @@ function (ParsedFileID $parsedFileID, Filesystem $fs, FileResolutionStrategy $st
                             // Move cached hash value to new location
                             $hasher->move($origin, $fs, $destination);
                         }
-                        $this->truncateDirectory(dirname($origin ?? ''), $fs);
+                        $tryTruncate[dirname($origin)] = true;
                     }
                 }
 
+                // Truncate any directories that held removed files
+                foreach (array_keys($tryTruncate) as $dir) {
+                    $this->truncateDirectory($dir, $fs);
+                }
```

Comment:
In almost 100% of cases all variants will be in the same directory, so we'll only have _one_ call here, instead of one per variant.

The only reason we still have to get an array of possible directories is because the way `findVariants()` and `FileIDHelper` are implemented allows for the possibility of custom implementations where variants could be in different directories.